.github/workflows/CI.yml

```
@@ -1,18 +1,22 @@
-name: Master Pull Request
+name: CI
 
 on:
   pull_request:
     branches: [master]
 
+concurrency: ci-${{ github.ref }}
+
 jobs:
-  validate:
+  format-lint-test-and-build:
```

```suggestion
  build:
```
What about just calling it `build`?

I think that's a bit better than repeating the step names in the job name, and the meaning of the process is building code that goes through defined steps. WDYT?

---

I wanted to be a bit more explicit and perhaps it makes sense to even split the job into its constituent parts at some point but I don't know, I would leave it as it is, it's not a big deal and at least keeps the CI step a little more descriptive until the job is split up, WDYT?

---

I don't mind if we leave it at that. It seems a bit odd, but if that's okay with you, I'm fine with it too :)